server/src/infra/repositories/job.repository.ts

```diff
@@ -140,7 +141,24 @@ export class JobRepository implements IJobRepository {
   }
 
   async queue(item: JobItem): Promise<void> {
-    await this.queueAll([item]);
+    const jobName = item.name;
+    const jobData = item.data ?? {};
+    const jobOptions = this.getJobOptions(item) || undefined;
+
+    // need to use add() instead of addBulk() for jobId deduplication
```

Comment:
This seems fine, but would it moving it into `queueAll`, which can look at `options.jobId` to determine which ones need to be added individually?

Reply:
Yeah, that's a good idea